packages/mobx-state-tree/src/types/complex-types/model.ts

```diff
@@ -117,65 +119,78 @@ export type ModelCreationType<T extends ModelProperties> = {
 } &
     { [K in keyof OptionalProps<T>]?: ExtractC<T[K]> }
 
+export type ModelCreationType2<T extends ModelProperties, FixedC> = FixedOrOther<
```

Comment:
At first I tried having the if inside ModelCreationType directly, but this ended up in types generation being messed up, therefore I had to split it. Either way the type generation will show/use ModelCreationType, not ModelCreationType2 (same for ModelSnapshotType)